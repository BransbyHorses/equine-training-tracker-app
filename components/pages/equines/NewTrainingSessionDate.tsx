import React from "react";
import {
	NewTrainingSessionContext,
	NewSkillTrainingSessionType,
} from "../../../utils/reducers/trainingSessionReducer";
import BackBreadcrumb from "../../BackBreadcrumb";
import ResponsiveButton from "../../ResponsiveButton";

const NewTrainingSessionDate = () => {
	const { state, dispatch } = React.useContext(NewTrainingSessionContext);

	console.log(state);

	return (
		<>
			<BackBreadcrumb />
			<ResponsiveButton
				onClick={() => dispatch({ type: NewSkillTrainingSessionType.NEXT, payload:  })}
			>
				Continue
			</ResponsiveButton>
		</>
	);
};

export default NewTrainingSessionDate;
