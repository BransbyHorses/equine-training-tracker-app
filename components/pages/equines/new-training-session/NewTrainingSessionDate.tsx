import React, { useEffect } from "react";
import {
	NewTrainingSessionContext,
	NewSkillTrainingSessionType,
} from "../../../../utils/reducers/trainingSessionReducer";

import ResponsiveButton from "../../../ResponsiveButton";
import BackBreadcrumb from "../../../BackBreadcrumb";
import PageTitle from "../../../PageTitle";

import { Box } from "@mui/material";
import TextField from "@mui/material/TextField";

import { MobileDatePicker } from "@mui/x-date-pickers/MobileDatePicker";
import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider";
import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";
import dayjs, { Dayjs } from "dayjs";

const NewTrainingSessionDate = () => {
	const {
		state: { newTrainingSession },
		dispatch,
	} = React.useContext(NewTrainingSessionContext);

	useEffect(() => {
		dispatch({
			type: NewSkillTrainingSessionType.SET_DATE,
			payload: dayjs(),
		});
	}, []);

	const changeDate = (newDate: Dayjs | null) => {
		dispatch({
			type: NewSkillTrainingSessionType.SET_DATE,
			payload: newDate,
		});
	};

	return (
		<>
			<BackBreadcrumb />
			<PageTitle title="What date was your training session?" />
			<Box sx={{ my: 4 }}>
				<LocalizationProvider dateAdapter={AdapterDayjs}>
					<MobileDatePicker
						label="Training Date"
						inputFormat="YYYY/MM/DD"
						value={newTrainingSession.date}
						onChange={changeDate}
						renderInput={(params) => <TextField {...params} fullWidth />}
					/>
				</LocalizationProvider>
			</Box>
			<ResponsiveButton
				disabled={!newTrainingSession.date}
				onClick={() => dispatch({ type: NewSkillTrainingSessionType.NEXT })}
			>
				Continue
			</ResponsiveButton>
		</>
	);
};

export default NewTrainingSessionDate;
