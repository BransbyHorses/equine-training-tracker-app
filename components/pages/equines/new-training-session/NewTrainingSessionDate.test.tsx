import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import NewTrainingSessionDate from "./NewTrainingSessionDate";

describe("NewTrainingSessionDate", () => {
	it("renders with minimum props", () => {
		const tree = renderer.create(<NewTrainingSessionDate />);
		expect(tree).toMatchSnapshot();
	});
});
