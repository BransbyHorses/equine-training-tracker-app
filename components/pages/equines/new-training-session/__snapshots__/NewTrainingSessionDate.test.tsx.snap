// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTrainingSessionDate renders with minimum props 1`] = `
[
  <nav
    className="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-7a84y9-MuiTypography-root-MuiBreadcrumbs-root"
  >
    <ol
      className="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
    >
      <li
        className="MuiBreadcrumbs-li"
      >
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-1qt0ygz-MuiTypography-root-MuiLink-root"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowLeftIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m14 7-5 5 5 5V7z"
            />
          </svg>
           Back
        </a>
      </li>
    </ol>
  </nav>,
  <div
    className="MuiBox-root css-17tcgb"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 css-1lni2c2-MuiTypography-root"
    >
      What date was your training session?
    </h4>
  </div>,
  <div
    className="MuiBox-root css-h5fkc8"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      onClick={[Function]}
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor=":r0:"
        id=":r0:-label"
      >
        Training Date
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-readOnly css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          aria-label="Choose date"
          aria-readonly={true}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-readOnly css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={true}
          required={false}
          type="text"
          value="2022/11/15"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-1ftyaf0"
          >
            <span>
              Training Date
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
  <button
    className="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1g0d0b8-MuiButtonBase-root-MuiButton-root"
    desktopstyles={
      {
        "mt": 3,
        "width": "20%",
      }
    }
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    Continue
  </button>,
]
`;
